plugins/backend/backstage-plugin-argo-cd-backend/src/plugin.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import { createRouter } from './service/router';
+import { Logger } from 'winston';
+
+export const ArgoCDPlugin = createBackendPlugin({
+  pluginId: 'argocd',
+  register(env) {
+    env.registerInit({
+      deps: {
+        http: coreServices.httpRouter,
+        logger: coreServices.logger,
+        config: coreServices.rootConfig,
+      },
+      async init({ http, logger, config }) {
+        logger.info('ArgoCD plugin is initializing');
+        http.use(
+          await createRouter({
+            logger: logger as Logger,
```

Comment:
This type is wrong. It's a `LoggerService` and not a `Winston.Logger`.

https://backstage.io/docs/reference/backend-plugin-api.loggerservice

If you want to preserve the `Winston.Logger` type, there's a util `@backstage/backend-common` to convert the `LoggerService`  to a `Winston.Logger`.

https://backstage.io/docs/reference/backend-common.loggertowinstonlogger

The usage I've seen is:

```ts
logger:  loggerToWinstonLogger(logger),
```